src/gas-price/gas-price.test.ts

```diff
@@ -373,12 +365,22 @@ describe(getRecommendedGasPrice.name, () => {
   });
 });
 
-describe(fetchAndStoreGasPrice.name, () => {
+describe(gasPriceModule.getGasPrice.name, () => {
+  it('returns gas price as bigint when internal rpc call returns a valid hex string', async () => {
+    jest.spyOn(provider, 'send').mockResolvedValueOnce('0x1a13b8600');
+    const gasPrice = await gasPriceModule.getGasPrice(provider);
+
+    expect(provider.send).toHaveBeenCalledWith('eth_gasPrice', []);
+    expect(gasPrice).toBe(BigInt('0x1a13b8600'));
+  });
+});
+
+describe(gasPriceModule.fetchAndStoreGasPrice.name, () => {
   it('fetches and stores the gas price from RPC provider', async () => {
     jest.spyOn(Date, 'now').mockReturnValue(dateNowMock);
-    jest.spyOn(provider, 'getFeeData').mockResolvedValueOnce({ gasPrice: ethers.parseUnits('10', 'gwei') } as any);
+    jest.spyOn(gasPriceModule, 'getGasPrice').mockResolvedValueOnce(ethers.parseUnits('10', 'gwei'));
```

Comment:
I believe you can spy on the provider instead and this way you avoid the need to import the entirety of gas price module. (Similarly in the test below).

Reply:
I see the point, but I think the ugliness it brings is not worth it. We can have a similarly readable test with:
```
jest.spyOn(provider, 'send').mockResolvedValueOnce(ethers.toBeHex(ethers.parseUnits('10', 'gwei')));
```